src/java.base/share/classes/sun/nio/cs/StreamDecoder.java

```diff
@@ -271,6 +273,26 @@ private boolean isOpen() {
         return !closed;
     }
 
+    public void fillZeroToPosition() throws IOException {
+        Object lock = this.lock;
+        if (lock instanceof InternalLock locker) {
+            locker.lock();
+            try {
+                lockedFillZeroToPosition();
+            } finally {
+                locker.unlock();
+            }
+        } else {
+            synchronized (lock) {
+                lockedFillZeroToPosition();
+            }
+        }
+    }
+
+    private void lockedFillZeroToPosition() throws IOException {
+        ensureOpen();
+        Arrays.fill(bb.array(), 0, bb.arrayOffset() + bb.position(), (byte)0);
```

Comment:
Should the `from` parameter be zero or `bb.arrayOffset()`?

Reply:
Right. Fixed.

Reply:
The buffer is allocated in StreamDecoder so I assume the array offset is 0 anyway, so I think they will be the same.